dev/README_RELEASE_AIRFLOW_UPGRADE_CHECK.md

```diff
@@ -0,0 +1,636 @@
+    ```
+
+- Generate SHA512/ASC (If you have not generated a key yet, generate it by following instructions on
+  http://www.apache.org/dev/openpgp.html#key-gen-generate-key)
+
+    ```shell script
+    ${AIRFLOW_REPO_ROOT}/dev/sign.sh apache-airflow-upgrade-check-${VERSION}-source.tar.gz
+    ${AIRFLOW_REPO_ROOT}/dev/sign.sh apache-airflow-upgrade-check-${VERSION}-bin.tar.gz
+    ${AIRFLOW_REPO_ROOT}/dev/sign.sh apache_airflow_upgrade_check-${VERSION}-py2.py3-none-any.whl
+    ```
+
+- Push the artifacts to ASF dev dist repo
+
+  ```shell script
+  # First clone the repo
+  svn checkout https://dist.apache.org/repos/dist/dev/airflow airflow-dev
+
+  # Create new folder for the release
+  cd airflow-dev/upgrade-check
+  svn mkdir ${VERSION}
+
+  # Move the artifacts to svn folder & commit
+  mv ${AIRFLOW_REPO_ROOT}/apache{-,_}airflow*-${VERSION}* ${VERSION}/
+  cd ${VERSION}
+  svn add *
+  svn commit -m "Add artifacts for Upgrade Check ${VERSION}"
+  ```
+
+## Prepare PyPI convenience "snapshot" packages
+
+At this point we have the artefact that we vote on, but as a convenience to developers we also want to
+publish "snapshots" of the RC builds to pypi for installing via pip. Also those packages
+are used to build the production docker image in DockerHub, so we need to upload the packages
+before we push the tag to GitHub. Pushing the tag to GitHub automatically triggers image building in
+DockerHub.
+
+To do this we need to
+
+- Build the package:
+
+    ```shell script
+    cd ${AIRFLOW_REPO_ROOT}
+    python -m setuptools_build_subpackage --subpackage-folder airflow/upgrade \
+        egg_info --tag-build "$(sed -e "s/^[0-9.]*//" <<<"$VERSION")" sdist \
+        --license-template ./LICENSE bdist_wheel
+    ```
+
+- Verify the artifacts that would be uploaded:
+
+    ```shell script
+    twine check dist/*
+    ```
+
+- Upload the package to PyPi's test environment:
+
+    ```shell script
+    twine upload -r pypitest dist/*
+    ```
+
+- Verify that the test package looks good by downloading it and installing it into a virtual environment.
+  The package download link is available at: https://test.pypi.org/project/apache-airflow-upgrade-check/#history
+
+- Upload the package to PyPi's production environment:
+
+  ```shell script
+  twine upload -r pypi dist/*
+  ```
+
+- Again, confirm that the package is available here: https://pypi.org/project/apache-airflow-upgrade-check
+
+It is important to stress that this snapshot should not be named "release", and it
+is not supposed to be used by and advertised to the end-users who do not read the devlist.
+
+- Push Tag for the release candidate
+
+    ```shell script
+    git push origin upgrade-check/${VERSION}
+    ```
+
+## Prepare Vote email on the Apache Airflow release candidate
+
+- Send out a vote to the dev@airflow.apache.org mailing list:
+
+Subject:
+
+```
+[VOTE] Release apache-airflow-upgrade-check 1.3.0 from 1.3.0rc1
```

Comment:
Maybe good to your VERSION from variable? I am not sure if we ever do it again but replacing the version is kinda easy to miss .